wgpu-hal/src/vulkan/mod.rs

```diff
@@ -149,6 +140,99 @@ pub struct Instance {
     shared: Arc<InstanceShared>,
 }
 
+/// The semaphores used to synchronize the swapchain image acquisition.
+///
+/// One of these is created per swapchain image and is used for the swapchain
+///
+/// The `acquire` semaphore is used to signal from the acquire operation to the first submit
+/// that uses the image.
+///
+/// The `present` semaphores are used to signal from each submit that uses the image to the
+/// present operation. We need use one per submit as we don't know at submit time which
+/// submission will be the last to use the image, so we add semaphores to them all.
+#[derive(Debug)]
+struct SwapchainSemaphores {
```

Comment:
Could we rename this `SwapchainImageSemaphores`, since it's per image, not per swapchain?

Reply:
Sure